README.md

```diff
@@ -1 +1,134 @@
-# stencil-progress-bar
+# Stencil Progress Bar
+
+A progress bar that comes in both standard and radial flavours. 
+
+[Demo Coming Soon](#)
+
+
+
+## Requirements
+
+- Sass
+- AMD loader
+- Dust templating
+
+
+
+## Installation
+
+Install the Stencil Progress Bar using the AdaptiveJS generator:
+```shell
+yo adaptivejs:component --install stencil-progress-bar
+```
+You can also install interactively by running `yo adaptivejs:component` and choosing the install option.
+
+
+
+## Usage
+
+Stencil components installed with the AdaptiveJS generator require no setup. Just load the component’s template in the desired view and use the dust helper to render it. For details, see [using Components in AdaptiveJS](https://github.com/mobify/devcenter-assets/blob/master/mobifydevcenter/v2.0/documentation/Components/Use%20a%20Component.md).
+
+
+
+## API
+
+### Template
+The template accepts the following parameters.
+
+Name | Type | Description
+-----|------|------------
+value | number within [0, 1] | Sets the initial value for the progress bar
+label | string | Sets the initial label for the progress bar
+state | one of ['inprogress', 'success', 'failure'] | Sets the initial state for the progress bar
+isRadial | boolean | If `true`, uses the radial progress bar instead of the linear progress bar
+
+This component has no template bodies.
+
+
+### UI Script
+
+First, require the Progress Bar UI script. Then call `ProgressBar.init`.
+
+```js
+require([
+    // ...
+    'progress-bar-ui'
+], function(
+    // ...
+    ProgressBar
+) {
+    ProgressBar.init($el);
+});
+```
+
+`ProgressBar.init` takes one parameter: `$el`.
+
+#### $el
+
+`$el` is the progress bar element. The component's instance is accessible through this DOM node. For instance, the following calls the `setProgress` method on an instance of the Progress Bar component:
+
+```js
+$('.c-progress-bar').data('progressbar').setProgress(0.5);
```

Comment:
@e Can you describe a use case for initializing more than one plugin on the same node?

The way I think of components is to imagine that they are custom HTML elements. It’s not possible to have say a `<select>` be something else at the same time. That’s the paradigm the framework has been working towards.